### ISSUE TITLE
CSS spelling and grammar features

### ISSUE THREAD

**Issue by delan** [mozilla/standards-positions#470]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title:
    * CSS Pseudo-Elements Module Level 4 (spelling and grammar features only)
    * CSS Text Decoration Module Level 4 (spelling and grammar features only)
* Specification or proposal URL:
    * https://drafts.csswg.org/css-pseudo-4/#selectordef-spelling-error
    * https://drafts.csswg.org/css-pseudo-4/#selectordef-grammar-error
    * https://drafts.csswg.org/css-text-decor-4/#valdef-text-decoration-line-spelling-error
    * https://drafts.csswg.org/css-text-decor-4/#valdef-text-decoration-line-grammar-error
* Caniuse.com URL (optional): 
    * https://caniuse.com/mdn-css_selectors_spelling-error
    * https://caniuse.com/mdn-css_selectors_grammar-error
    * none for the new text-decoration-line values yet
* Bugzilla URL (optional): none yet
* Mozillians who can provide input (optional): unsure

### Other information

While these features span two specs, they might be worth considering together, because the new pseudo-elements depend on the new text-decoration-line values for UA stylesheet support (and to be most useful to authors).

**Comment by emilio** [comment]
cc @jfkthame

**Comment by emilio** [comment]
One question, is the intention for the matching / inheritance behavior to match ::selection? And if so, the current browser behavior or the behavior that is in the spec, which is significantly different?

**Comment by delan** [comment]
> One question, is the intention for the matching / inheritance behavior to match ::selection? And if so, the current browser behavior or the behavior that is in the spec, which is significantly different?

Yes, the processing model would be the same as ::selection. I didn’t realise at the time that the current spec behaviour wasn’t implemented by anyone yet^. I think it would be ok to start prototyping under the current impl behaviour (because they would still be usable enough to play around with), but reconcile the gap (changing either spec or impl) before we ship. That might mean gating the pseudos behind a flag, to avoid adding to the backwards-compatibility risk we already have with ::selection.

^ I would really appreciate some of your time to ask you about that actually. I spent a few days [reading how the behaviour evolved](https://twitter.com/dazabani/status/1349436774515281920), but a lot of the terminology was new to me, so I’m still not sure I’ve understood the nuances _correctly_. If you’re interested, let me know what medium works best for you. Thanks!

**Comment by delan** [comment]
I’ve created a separate issue (#548) about the highlight processing model.
